impl/src/main/java/org/jboss/weld/xml/BeansXmlParser.java

```diff
@@ -178,11 +196,12 @@ private boolean alreadyAdded(Metadata<String> metadata, List<Metadata<String>> l
         return false;
     }
 
-    private static InputSource[] loadXsds() {
+    private static InputSource[] loadXsds(String beansXmlAsString) {
 
         List<InputSource> xsds = new ArrayList<InputSource>();
+        String beans11 = "beans_1_1.xsd";
 
-        for (XmlSchema schema : XmlSchema.values()) {
+        for (XmlSchema schema : XmlSchema.getSchemas(beansXmlAsString != null && beansXmlAsString.contains(beans11))) {
```

Comment:
The `beans11` variable is not needed - we could use `XmlSchema.CDI11.getFileName()` instead.